File: python/questions/canJump.py
from typing import List


class Solution:
    def canJump(self, nums: List[int]) -> bool:
        ln = len(nums)
        ms = 0
        i = 0
        while i <= ms:
            ms = max(ms, i + nums[i])
            i += 1
            if ms >= ln - 1:
                return True
        return False

File: cpp/questions/canJump.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool canJump(vector<int>& nums) {
        int ns = nums.size();
        int mp = 0;
        for (int i=0; i<=mp; ++i) {
            mp = std::max(mp, i+nums[i]);
            if (mp >= ns-1) {
                return true;
            }
        }
        return false;
    }
};
